api/src/lib.rs

```diff
@@ -133,3 +137,52 @@ pub async fn initialize_db() -> Result<SqliteConnection, Box<dyn Error>> {
 
     Ok(connection)
 }
+
+pub struct PostieDb {
+    pub connection: SqliteConnection
+}
+
+#[derive(Debug)]
+pub struct DBRequest {
+    id: String,
+    method: String,
+    url: String,
+    name: String,
+    headers: String,
+    body: String
+}
+
+struct DBResponse {
+    id: String,
+    status_code: u8,
+    name: String,
+    headers: String,
+    body: String
+}
+
+struct DBRequestHistoryItem {
+    id: String,
+    request_id: String,
+    response_id: String,
+    sent_at: String,
+    response_time: u32
+}
+
+impl PostieDb {
+    pub async fn new() -> Self {
+        PostieDb { connection: initialize_db().await.ok().unwrap() }
+    }
+
+    pub async fn save_request_history(&mut self, request: DBRequest) -> Result<(), Box<dyn Error>> {
+        println!("got request: {:?}", request);
+        let mut transaction = self.connection.begin().await?;
+        // let res = sqlx::query!("SELECT * FROM request_history").fetch_all(&mut *transaction).await;
+
+        let _request = sqlx::query_as!(DBRequest, "INSERT INTO request (id, method, url, name, headers, body) VALUES ($1, $2, $3, $4, $5, $6)", request.id, request.method, request.url, request.name, request.headers, request.body).execute(&mut *transaction).await?;
```

Comment:
Not AS concise as some ORM libs in node land like we're used to (mainly since sql isn't really trying to be an ORM) but having compile time query checking is pretty snazzy imo lol.

Looking through the sql docs it seems like there are actual ORM crates that's are build on top on sqlx? https://github.com/SeaQL/sea-orm looks pretty decent for the future maybe?